### PR TITLE
Remove debug information globals

### PR DESCRIPTION
**Pull request by pchintalapudi**
Another PR in the line of removing the codegen globals.

Depends on #43802

### CODE REVIEW

**Comment by vtjnash** [comment]
Doesn't this need to get shared among all `jl_codegen_params_t`, so that that these only get created once for each Module?

**Comment by pchintalapudi** [comment]
What happens if we create the debug information builders multiple times? Is it just a performance loss, or will LLVM silently break debuginfo?

**Comment by vtjnash** [comment]
It shouldn't horribly fail, since the types don't have to agree, but will probably make unique types that are supposed to be the same.

File: src/Makefile [hunk 1]
```diff
@@ -297,6 +297,7 @@ $(BUILDDIR)/jltypes.o $(BUILDDIR)/jltypes.dbg.obj: $(SRCDIR)/builtin_proto.h
 $(build_shlibdir)/libllvmcalltest.$(SHLIB_EXT): $(SRCDIR)/codegen_shared.h $(BUILDDIR)/julia_version.h
 $(BUILDDIR)/llvm-alloc-helpers.o $(BUILDDIR)/llvm-alloc-helpers.dbg.obj: $(SRCDIR)/codegen_shared.h $(SRCDIR)/llvm-pass-helpers.h $(SRCDIR)/llvm-alloc-helpers.h
 $(BUILDDIR)/llvm-alloc-opt.o $(BUILDDIR)/llvm-alloc-opt.dbg.obj: $(SRCDIR)/codegen_shared.h $(SRCDIR)/llvm-pass-helpers.h $(SRCDIR)/llvm-alloc-helpers.h
+$(BUILDDIR)/llvm-cpufeatures.o $(BUILDDIR)/llvm-cpufeatures.dbg.obj: $(SRCDIR)/jitlayers.h
```
Comment (vchuravy) [inline src/Makefile:300]: Can you add this as a separate PR since we need to backport this.

File: src/aotcompile.cpp [hunk 2]
```diff
@@ -453,11 +453,11 @@ void jl_dump_native_impl(void *native_code,
     TheTriple.setOS(llvm::Triple::MacOSX);
 #endif
     std::unique_ptr<TargetMachine> TM(
-        jl_TargetMachine->getTarget().createTargetMachine(
+        jl_ExecutionEngine->getTargetMachine().getTarget().createTargetMachine(
```
Comment (vchuravy) [inline src/aotcompile.cpp:456]: this seems a little bit odd. Should we use a TargetMachineBuilder?

File: src/Makefile (hunk 1 above)
Comment (pchintalapudi) [reply src/Makefile:300]: Actually this include is new to this PR, so it shouldn't have to be backported right?

File: src/aotcompile.cpp (hunk 2 above)
Comment (pchintalapudi) [reply src/aotcompile.cpp:456]: We probably should, especially since we can choose to not expose the actual TargetMachine itself from the JIT engine.

Comment (pchintalapudi) [reply src/aotcompile.cpp:457]: Update on this: my attempt at using a TargetMachineBuilder failed during building on macos, so I'm thinking it's better to revisit rewriting this particular sequence of instructions at a later date.

**Pull request merged**